src/pythermalcomfort/models.py

```diff
@@ -1082,60 +1084,56 @@ def exponential(t_db):
             (-1.8680009 * (10 ** (-13))),
         ]
         tk = t_db + 273.15  # air temp in K
-        es = 2.7150305 * math.log1p(tk)
+        es = 2.7150305 * np.log1p(tk)
         for count, i in enumerate(g):
             es = es + (i * (tk ** (count - 2)))
-        es = math.exp(es) * 0.01  # convert Pa to hPa
+        es = np.exp(es) * 0.01  # convert Pa to hPa
         return es
 
     # Do a series of checks to be sure that the input values are within the bounds
     # accepted by the model.
-    if (
-        (tdb < -50.0)
-        or (tdb > 50.0)
-        or (tr - tdb < -30.0)
-        or (tr - tdb > 70.0)
-        or (v < 0.5)
-        or (v > 17)
-    ):
-        raise ValueError(
-            "The value you entered are outside the equation applicability limits"
-        )
+    tdb_valid = valid_range(tdb, (-50.,50.))
+    diff_valid = valid_range(tr - tdb, (-30.,70.))
+    v_valid = valid_range(v, (0.5, 17.))
+    all_valid = ~(np.isnan(tdb_valid) | np.isnan(diff_valid) | np.isnan(v_valid))
 
     eh_pa = exponential(tdb) * (rh / 100.0)
     delta_t_tr = tr - tdb
     pa = eh_pa / 10.0  # convert vapour pressure to kPa
 
     utci_approx = utci_optimized(tdb, v, delta_t_tr, pa)
 
-    if utci_approx < -40:
-        stress_category = "extreme cold stress"
-    elif utci_approx < -27:
-        stress_category = "very strong cold stress"
-    elif utci_approx < -13:
-        stress_category = "strong cold stress"
-    elif utci_approx < 0:
-        stress_category = "moderate cold stress"
-    elif utci_approx < 9:
-        stress_category = "slight cold stress"
-    elif utci_approx < 26:
-        stress_category = "no thermal stress"
-    elif utci_approx < 32:
-        stress_category = "moderate heat stress"
-    elif utci_approx < 38:
-        stress_category = "strong heat stress"
-    elif utci_approx < 46:
-        stress_category = "very strong heat stress"
-    else:
-        stress_category = "extreme heat stress"
-
-    if units.lower() == "ip":
```

Comment:
Have you removed this function? We need to convert back the results into IP units if the user selected them.